templates/tree-editor/tree/tree-model-service.ts

```diff
@@ -2,81 +2,72 @@ import { ILogger } from '@theia/core';
 import { inject, injectable } from 'inversify';
 import { TreeEditor } from 'theia-tree-editor';
 
-import { TreeModel } from './tree-model';
-import { 
-  schema,
-  personView,
-  componentView
+import { CoffeeModel } from './tree-model';
+import {
+    brewingView,
+    coffeeSchema,
+    controlUnitView,
+    dripTrayView,
+    machineView,
+    waterTankView,
 } from './tree-schema';
 
 @injectable()
 export class TreeModelService implements TreeEditor.ModelService {
 
-  constructor(@inject(ILogger) private readonly logger: ILogger) { }
+    constructor(@inject(ILogger) private readonly logger: ILogger) { }
 
-  getDataForNode(node: TreeEditor.Node) {
-    return node.jsonforms.data;
-  }
-
-  getSchemaForNode(node: TreeEditor.Node) {
-    return {
-      definitions: schema.definitions,
-      ...this.getSubSchemaForNode(node)
-    };
-  }
-
-  private getSubSchemaForNode(node: TreeEditor.Node) {
-    const schema = this.getSchemaForType(node.jsonforms.data.typeId);
-    if (schema) {
-      return schema;
+    getDataForNode(node: TreeEditor.Node) {
+        return node.jsonforms.data;
     }
-    return undefined;
-  }
 
-  private getSchemaForType(type: string) {
-    if (!type) {
-      return undefined;
-    }
-    const localSchema = Object.entries(schema.definitions)
-      .map(entry => entry[1])
-      .find(
-        definition =>
-          definition.properties && definition.properties.typeId.const === type
-      );
-    if (!localSchema) {
-      this.logger.warn("Can't find definition schema for type " + type);
-    }
-    return localSchema;
-}
-  
-  getUiSchemaForNode(node: TreeEditor.Node) {
-    const schema = this.getUiSchemaForType(node.jsonforms.data.typeId);
-    if (schema) {
-      return schema;
+    getSchemaForNode(node: TreeEditor.Node) {
+        return {
+            definitions: coffeeSchema.definitions,
+            ...this.getSchemaForType(node.jsonforms.type),
+        };
     }
-    return undefined;
-  }
 
-  private getUiSchemaForType(type: string) {
-    if (!type) {
-      return undefined;
+    private getSchemaForType(type: string) {
+        if (!type) {
+            return undefined;
+        }
+        const schema = Object.entries(coffeeSchema.definitions)
+            .map(entry => entry[1])
+            .find(
+                definition =>
+                    definition.properties && definition.properties.typeId.const === type
+            );
+        if (!schema) {
```

Comment:
Theoretically a valid schema would also be `false`. Now this is pretty minor but we could still handle it easily
```suggestion
        if (schema === undefined) {
```